reconcile/saas_auto_promotions_manager/subscriber.py

```diff
@@ -142,22 +144,42 @@ def _compute_desired_ref(self) -> None:
                     break
                 publisher_refs.add(publisher.commit_sha)
 
+        # By default we keep current state
+        self.desired_ref = self.ref
+
         if len(publisher_refs) > 1:
             logging.info(
                 "Publishers for subscriber at path %s have mismatching refs: %s",
                 self.target_file_path,
                 publisher_refs,
             )
-        if (
-            len(publisher_refs) != 1
-            or any_bad_deployment
-            or not self._passed_accumulated_soak_days()
-        ):
-            # We keep current state
-            self.desired_ref = self.ref
-        else:
-            # We have a common single publisher ref w/o any deployment issues
-            self.desired_ref = next(iter(publisher_refs))
+            return
+
+        if not self._passed_accumulated_soak_days():
+            logging.debug(
+                "Subscriber at path %s promotion stopped because of soak days",
+                self.target_file_path,
+            )
+            return
+
+        if any_bad_deployment:
+            logging.info(
+                "Subscriber at path %s promotion stopped because of bad publisher deployment",
+                self.target_file_path,
+            )
+            return
+
+        desired_ref = next(iter(publisher_refs))
```

Comment:
in old code, we have `len(publihser_refs) != 1` check, we checked `> 1` case above, can there be any case `publisher_refs` is empty?

Reply:
Good catch! Added the check :+1: